services/notify/src/irc.js

```diff
@@ -70,69 +69,47 @@ class IRCBot {
       }
     });
 
-    let queueUrl = await this.sqs.createQueue({
-      QueueName: this.queueName,
-    }).promise().then(req => req.QueueUrl);
-
-    this.done = (async () => {
-      debug('Connecting to: ' + queueUrl);
-      while (!this.stopping) {
-        debug('Waiting for message from sqs.');
-        let req = await this.sqs.receiveMessage({
-          QueueUrl: queueUrl,
-          AttributeNames: ['ApproximateReceiveCount'],
-          MaxNumberOfMessages: 10,
-          VisibilityTimeout: 30,
-          WaitTimeSeconds: 20,
-        }).promise();
-        if (!req.Messages) {
-          debug('Did not receive any messages from sqs in timeout.');
-          continue;
-        }
-        debug(`Received ${req.Messages.length} messages from sqs.`);
-        let success = 0;
-        for (let message of req.Messages) {
-          try {
-            await this.notify(JSON.parse(message.Body));
-          } catch (err) {
-            console.log('Failed to send IRC notification: %j, %s', err, err.stack);
-            // Skip deleting if we're below MAX_RETRIES
-            if (message.Attributes.ApproximateReceiveCount < MAX_RETRIES) {
-              continue;
-            }
-          }
-          // Delete message
-          await this.sqs.deleteMessage({
-            QueueUrl: queueUrl,
-            ReceiptHandle: message.ReceiptHandle,
-          }).promise();
-          success += 1;
-        }
-        debug(`Deleted ${success} message from sqs.`);
-      }
-      debug('Stopping irc sqs loop');
-    })();
+    this.pq = await consume({
+      client: this.pulseClient,
+      bindings: [{exchange: 'irc-notification', routingKeyPattern: 'irc'}],
+      queueName: 'irc-notifications',
+    },
+    this.monitor.timedHandler('notification', this.onMessage.bind(this))
+    );
   }
 
-  async notify({channel, user, message}) {
+  async onMessage({payload}) {
+    let {channel, user, message} = payload.message;
+    let retires = 0;
     if (channel && !/^[#&][^ ,\u{0007}]{1,199}$/u.test(channel)) {
       debug('irc channel ' + channel + ' invalid format. Not attempting to send.');
       return;
     }
     debug(`Sending message to ${user || channel}: ${message}.`);
-    if (channel) {
-      // This callback does not ever have an error. If it triggers, we have succeeded
-      // Time this out after 10 seconds to avoid blocking forever
-      await new Promise((accept, reject) => {
-        setTimeout(() => {
-          debug('Timed out joining channel, may be ok. Proceeding.');
-          accept();
-        }, 10000);
-        this.client.join(channel, accept);
-      });
+    while(true) {
```

Comment:
We can actually simplify this I think. Since [pulse consumers](https://github.com/taskcluster/taskcluster/tree/master/libraries/pulse#pulseconsumer) will retry a single time when a handler fails, we can just fail the handler when something goes wrong and hope that the single retry fixes things. This irc stuff is best effort for the most part so once is enough. We can get rid of any explicit retry logic in here including `MAX_RETRIES`. Does that make sense?